include/psa/crypto_extra.h

```diff
@@ -121,6 +121,14 @@ psa_status_t mbedtls_psa_inject_entropy(const unsigned char *seed,
                                         size_t seed_size);
 
 
+/* The Number of key slots (plus one because 0 is not used).
+ * The value is a compile-time constant for now, for simplicity. */
+#if defined(MBEDTLS_PSA_KEY_SLOT_COUNT)
+#else
+    #define MBEDTLS_PSA_KEY_SLOT_COUNT 32
```

Comment:
Hi @gilles-peskine-arm, 
regarding adding this flag to config.h, you wrote in the email chain: 
"I'm not keen on adding it to mbedtls/config.h because it isn't necessarily something that we'll maintain backward compatibility with. "

This is the reason it was left out, I can add it now if you prefer

Reply:
Hmm, yes, that is a good point, we aren't going to promise backward compatibility for this, so it shouldn't be in `config.h`, or at least not until we revise (and document!) our backward compatibility rules. So please keep it in `crypto_extra.h`, and add a comment that states that this is an implementation detail that may change or be removed in future versions. By default we _do_ promise to maintain backward compatibility with things defined in `crypto_extra.h`.

Reply:
comment added

Reply:
We do have precedents for defining options in `config.h` that we don't intend to preserve backward compatibility for: `MBEDTLS_PSA_HAS_ITS_IO`, `MBEDTLS_PSA_CRYPTO_SPM`, the PSA storage modules. We intend to preserve the functionality, but not necessarily the configuration interface.

@Patater What's your opinion on this?

Reply:
More generally, in Mbed Crypto, we don't intend to preserve compatibility with `MBEDTLS_xxx` APIs in the long term. So I think my view of backward compatibility is behind the times and the configuration option should go in `config.h`.

Reply:
@gilles-peskine-arm and @Patater , what is the final ruling on this? (add to config.h or not)